homeassistant/components/hunterdouglas_powerview/config_flow.py

```diff
@@ -4,80 +4,162 @@
 import logging
 
 from aiopvapi.helpers.aiorequest import AioRequest
+from aiopvapi.hub import Hub
 import async_timeout
 import voluptuous as vol
 
 from homeassistant import config_entries, core, exceptions
 from homeassistant.components import dhcp, zeroconf
-from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.const import CONF_API_VERSION, CONF_HOST, CONF_NAME
 from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import selector
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
 
 from . import async_get_device_info
 from .const import DOMAIN, HUB_EXCEPTIONS
 
 _LOGGER = logging.getLogger(__name__)
 
-DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
+API_VERSION_AUTO = "Detect Automatically"
+API_VERSION_1 = "Generation 1"
+API_VERSION_2 = "Generation 2"
+API_VERSION_3 = "Generation 3"
+API_SCHEMA: list[str] = [API_VERSION_AUTO, API_VERSION_1, API_VERSION_2, API_VERSION_3]
+API_MAP: dict[str, int | None] = {
+    API_VERSION_AUTO: None,
+    API_VERSION_1: 1,
+    API_VERSION_2: 2,
+    API_VERSION_3: 3,
+}
```

Comment:
Many users will not know which hub version they will have. We should always detect it for them

Reply:
yea i considered that as an issue too - was just trying to consider a bad detect from the upstream api (gen1 v gen2)
But that was why i made it an advanced feature (made testing the new is_supported easy too :wink:) 

If we jsut want to keep it simple i can remove all that and auto detect